Apps/phsmime/smime_connector.py

```diff
@@ -0,0 +1,508 @@
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            return action_result.set_status(phantom.APP_ERROR, "Error occurred while adding data to the 'action_result'. {}".format(err))
+
+        # Return success, no need to set the message, only the status
+        # BaseConnector will create a textual message based off of the summary dictionary
+        return action_result.set_status(phantom.APP_SUCCESS, SMIME_VERIFY_OK_MSG)
+
+    def handle_action(self, param):
+
+        ret_val = phantom.APP_SUCCESS
+
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'sign_email':
+            ret_val = self._handle_sign_email(param)
+        elif action_id == 'encrypt_email':
+            ret_val = self._handle_encrypt_email(param)
+        elif action_id == 'decrypt_email':
+            ret_val = self._handle_decrypt_email(param)
+        elif action_id == 'verify_email':
+            ret_val = self._handle_verify_email(param)
+
+        return ret_val
+
+    def initialize(self):
+
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+
+        config = self.get_config()
+
+        # Fetching the Python major version
+        try:
+            self._python_version = int(sys.version_info[0])
+        except:
+            return self.set_status(phantom.APP_ERROR, "Error occurred while getting the Phantom server's Python major version")
+
+        # Load keys
+        self._keys = {
+            "private": self._handle_py_ver_compat_for_input_str(config.get('private_key')),
+            "public": self._handle_py_ver_compat_for_input_str(config.get('public_key'))
+        }
+
+        return phantom.APP_SUCCESS
+
+    def finalize(self):
+
+        # Save the state, this data is saved across actions and app upgrades
+        self.save_state(self._state)
+        return phantom.APP_SUCCESS
+
+
+if __name__ == '__main__':
+
+    import pudb
+    import argparse
+
+    pudb.set_trace()
+
+    argparser = argparse.ArgumentParser()
+
+    argparser.add_argument('input_test_json', help='Input Test JSON file')
+    argparser.add_argument('-u', '--username', help='username', required=False)
+    argparser.add_argument('-p', '--password', help='password', required=False)
+
+    args = argparser.parse_args()
+    session_id = None
+
+    username = args.username
+    password = args.password
+
+    if (username is not None and password is None):
+
+        # User specified a username but not a password, so ask
+        import getpass
+        password = getpass.getpass("Password: ")
+
+    if (username and password):
+        try:
+            login_url = SmimeConnector._get_phantom_base_url() + '/login'
+            print ("Accessing the Login page")
```

Comment:
nit: can we remove the space between `print` and `(`? same for later uses

Reply:
Added the suggested change in this commit: https://github.com/phantomcyber/phantom-apps/pull/365/commits/66e66e3db9aba57c4758ff60b82751d9ce57a8b4